### PR TITLE
Enhanced the get_support_bundle script

### PR DESCRIPTION
**Pull request by phrtp**
Enhanced the get_support_bundle script. Added the ability to specify pod roles to be collected, which will cut down on  bundle size. Also added logic to gather Cassandra and Elasticsearch storage information, as well as container density.

### CODE REVIEW

File: scripts/get_support_bundle.sh [hunk 1]
```diff
@@ -1,25 +1,94 @@
...
+
 NAMESPACE=${1:-sysdigcloud}
-CONTEXT=${2:-}
+LABELS=""
+LOG_DIR=$(mktemp -d sysdigcloud-support-bundle-XXXX)
 
-if [[ -z ${CONTEXT} ]] ; then
-    context=""
-else
-    context="--context ${CONTEXT}"
```
Comment (mbreitung) [inline scripts/get_support_bundle.sh:12]: I don't think that all the work around using `--context` in all kubectl commands should be removed. Personally I never `set-context` in my shell as I am consistently managing multiple clusters

File: scripts/get_support_bundle.sh [hunk 2]
```diff
@@ -1,25 +1,105 @@
...
+	printf "Pods: ${num_pods}\n" >> ${LOG_DIR}/container_density.txt
+	printf "Running Containers: ${num_running_containers}\n" >> ${LOG_DIR}/container_density.txt
+	printf "Containers: ${num_total_containers}\n" >> ${LOG_DIR}/container_density.txt
+
+    
+  esac
+done
+
+if [[ -z ${NAMESPACE} ]]; then
+  NAMESPACE=sysdigcloud
```
Comment (mbreitung) [inline scripts/get_support_bundle.sh:82]: namespace is already defaulted 2x above. I don't think that this is needed

File: scripts/get_support_bundle.sh [hunk 3]
```diff
@@ -1,25 +1,105 @@
...
+fi
+
+if [[ -z ${CONTEXT} ]]; then
+  CONTEXT=""
 fi
 
 #verify that the provided namespace exists
-kubectl ${context} get namespace ${NAMESPACE} > /dev/null
+kubectl get namespace ${NAMESPACE} > /dev/null
```
Comment (mbreitung) [inline scripts/get_support_bundle.sh:90]: this command needs the context

File: scripts/get_support_bundle.sh [hunk 4]
```diff
@@ -1,44 +1,128 @@
...
+			echo "-d  Include container density information"; printf "\n";     
+			exit;;
+
+		C) CONTEXT=${OPTARG:-};;
+
+		c) echo "Fetching Cassandra storage info";
+
+			# Executes a df -h in Cassandra pod, gets proxyhistograms, tpstats, and compactionstats
+			printf "Pod#\tFilesystem\tSize\tUsed\tAvail\tUse\tMounted on\n" > ${LOG_DIR}/cassandra_storage.log
+			for pod in $(kubectl get pods -l role=cassandra -n ${NAMESPACE} | grep -v "NAME" | awk '{print $1}')
```
Comment (mbreitung) [inline scripts/get_support_bundle.sh:31]: this is missing the context flag

File: scripts/get_support_bundle.sh [hunk 5]
```diff
@@ -1,44 +1,128 @@
...
+			do
+				#printf "$pod\t" > ${LOG_DIR}/cassandra_storage.log
+				kubectl exec -it $pod -n ${NAMESPACE} -- df -Ph | grep cassandra | grep -v "tmpfs" | awk '{printf "%-13s %10s %6s %8s %6s %s\n",$1,$2,$3,$4,$5,$6}' > ${LOG_DIR}/cassandra_storage.log
+			done
+
+			for pod in $(kubectl get pods -l role=cassandra -n sysdig | grep -v "NAME" | awk '{print $1}'); 
+			do 
+				echo $pod; for cmd in proxyhistograms status tpstats compactionstats; do kubectl exec $pod -c cassandra -n sysdig -- nodetool $cmd > ${LOG_DIR}/cassandra_nodes_output.log;
```
Comment (mbreitung) [inline scripts/get_support_bundle.sh:39]: please expand this line

File: scripts/get_support_bundle.sh [hunk 6]
```diff
@@ -1,44 +1,128 @@
...
-kubectl ${context} get namespace ${NAMESPACE} > /dev/null
+kubectl --context=${CONTEXT} get namespace ${NAMESPACE} > /dev/null
 
-KUBE_OPTS="--namespace ${NAMESPACE} ${context}"
+KUBE_OPTS="--namespace ${NAMESPACE} --context=${CONTEXT}"
 
-LOG_DIR=$(mktemp -d sysdigcloud-support-bundle-XXXX)
 
-SYSDIGCLOUD_PODS=$(kubectl ${KUBE_OPTS} get pods | awk '{ print $1 }' | grep -v NAME)
+if [[ -z ${LABELS} ]]; then
```
Comment (mbreitung) [inline scripts/get_support_bundle.sh:95]: for/do is on separate lines. Please use the same pattern for all blocks

File: scripts/get_support_bundle.sh [hunk 7]
```diff
@@ -1,44 +1,128 @@
...
+	NAMESPACE=sysdigcloud
+fi
+
+if [[ -z ${CONTEXT} ]]; then
+	CONTEXT=""
 fi
 
 #verify that the provided namespace exists
-kubectl ${context} get namespace ${NAMESPACE} > /dev/null
+kubectl --context=${CONTEXT} get namespace ${NAMESPACE} > /dev/null
```
Comment (mbreitung) [inline scripts/get_support_bundle.sh:90]: if context is not set, then this line will fail

File: scripts/get_support_bundle.sh [hunk 8]
```diff
@@ -1,44 +1,128 @@
...
+			printf "Nodes: ${num_nodes}\n" >> ${LOG_DIR}/container_density.txt
+			printf "Pods: ${num_pods}\n" >> ${LOG_DIR}/container_density.txt
+			printf "Running Containers: ${num_running_containers}\n" >> ${LOG_DIR}/container_density.txt
+			printf "Containers: ${num_total_containers}\n" >> ${LOG_DIR}/container_density.txt
+
+    
+	esac
+done
+
+if [[ -z ${NAMESPACE} ]]; then
```
Comment (mbreitung) [inline scripts/get_support_bundle.sh:81]: namespace and context are already defaulted above. These 2 if block are not needed

File: scripts/get_support_bundle.sh [hunk 9]
```diff
@@ -1,44 +1,128 @@
...
+if [[ -z ${CONTEXT} ]]; then
+	CONTEXT=""
 fi
 
 #verify that the provided namespace exists
-kubectl ${context} get namespace ${NAMESPACE} > /dev/null
+kubectl --context=${CONTEXT} get namespace ${NAMESPACE} > /dev/null
 
-KUBE_OPTS="--namespace ${NAMESPACE} ${context}"
+KUBE_OPTS="--namespace ${NAMESPACE} --context=${CONTEXT}"
```
Comment (mbreitung) [inline scripts/get_support_bundle.sh:92]: if context is not set, then this line will cause all kubectl lines to fail

**Comment by patrickhargett** [comment]
@mbreitung does this look okay? Can we merge this?

**Comment by patrickhargett** [comment]
We will move forward by using the sysdig-cloud-scripts repo.

**Pull request closed without merging**